### DISCUSSION TITLE
How to wait for the output of a watch mode dependency

### DISCUSSION THREAD

**Discussion by c-kirkeby** [vercel/turborepo#1549] [Help] [answered]
Our setup is pretty simple; we've got at least two services that all run in watch mode with the command `start:dev`:

> turbo.json
```json
{
  "$schema": "https://turborepo.org/schema.json",
  "baseBranch": "origin/main",
  "pipeline": {
    "build": {
      "dependsOn": ["^build"],
      "outputs": ["dist/**", ".next/**"]
    },
    "test": {},
    "deploy": {
      "dependsOn": ["build", "test", "lint"]
    },
    "lint": {},
    "start:dev": {
      "cache": false
    },
    "start": {
      "dependsOn": ["build"]
    },
    "migration:run": {
      "cache": false
    },
    "fixture:run": {
      "cache": false
    }
  }
}
```

`ui` depends is a next.js app which depends on a vite react library called `components`. The library spits out three relevant files for dist:
```
lib.es.js
lib.umd.js
styles.css
```
Each rebuild deletes and recreates the styles.css bundle, causing a race condition as ui needs to rebuild after detecting a change in the dependencies, and the styles.css bundle being recreated.

I added the following line hoping it would allow them to rebuild in a sequential manner, but now `ui` just never runs:

```
    "ui#start:dev": {
      "cache": false,
      "dependsOn": ["components#start:dev"]
    },
```

Not sure how to fix this - any help would be appreciated.

**Comment by rafaeltab** [comment, marked as answer]
I am afraid no such thing exists currently, you could always not delete the files and only ever replace them.
Maybe [typescript projects](https://www.typescriptlang.org/docs/handbook/project-references.html) could help you here? I don't have enough experience with next.js to tell.

Related issues: 

1. https://github.com/vercel/turborepo/issues/460
2. https://github.com/vercel/turborepo/issues/725

**Reply by c-kirkeby** [reply]
Vite controls the reanimation of the `dist` files. It might be something I can configure in vite if there's not a solution in turbo for sequential watch actions.
